src/ViewRendererBundle/objects/helper/TPkgViewRendererSnippetDirectory.class.php

```diff
@@ -408,6 +408,22 @@ public function getBasePaths($oPortal = null, $sBaseDirectory = null)
             }
         }
 
+        if (true === $this->requestInfoService->isBackendMode()) {
+            $allBundles = $this->kernel->getBundles();
```

Comment:
In the interest of keeping methods easy to read, it would probably make sense to extract this block of code into a method, so it becomes something like

```php
if (true === $this->requestInfoService->isBackendMode()) {
    $aBasePaths[$sPortalCacheId] = array_merge(
        $aBasePaths[$sPortalCacheId],
        $this->getBasePathsForBackendModules()
    );
}
```

Reply:
Move (all) to a method.